src/herder/HerderImpl.cpp

```diff
@@ -1047,6 +1047,25 @@ HerderImpl::persistSCPState(uint64 slot)
         }
     }
 
+    // Remember EXTERNALIZE envelopes for previous slots as well
+    auto prevSlot = slot - 1;
+    while (prevSlot > 0 && slot - prevSlot < Herder::MAX_SLOTS_TO_REMEMBER)
```

Comment:
I think this is wrong: we need to do the same thing than in the previous loop (remember `txset` and `qset` values). We probably need to add a test case that old slots are indeed properly persisted.

Are there performance implications by implementing it like this?
it's going to save 12 transaction sets in the same blob object (potentially over 100Mb of data) every time we call `emitEnvelope` (several times per externalize).

I suspect we should:
* at a minimum only update the data for old slots when we externalize
* better, use some sort of "circular data structure", that allows to only update one of the old persisted slots (by, for example, using not just 1 entry `kLastSCPData` but 13 separate)